include/fplus/side_effects.hpp

```diff
@@ -283,6 +283,46 @@ auto execute_n_times(std::size_t n, const Effect& eff)
     }
 }
 
+// API search type: parallel_for : (Io a, [a]) -> Io ()
+// Runs the function `f` in parallel on all the `idxs_to_iterate` container elements.
+// The function will perform its side effects, and nothing is returned.
+// -
+// Example:
+// -
+// constexpr int rows = 3000, cols = 2000;
+// MandelbrotMat<rows, cols> mandelbrot_mat;
+// auto fill_mandelbrot_row = [& mandelbrot_mat ](int row_idx) {
+//        ...
+// };
+// mandelbrot_mat = MandelbrotMat<rows, cols>();
+// auto rows_idxs = fplus::numbers(0, rows);
+// fplus::parallel_for(fill_mandelbrot_row, rows_idxs);
+template<typename F, typename ContainerIdxs>
+void parallel_for(F f, const ContainerIdxs& idxs_to_iterate)
+{
+    using IdxType = typename ContainerIdxs::value_type;
+    auto f_dummy_return = [&f](const IdxType& v) {
+        f(v);
+        return true;
+    };
+    fplus::transform_parallelly(f_dummy_return, idxs_to_iterate);
+}
+
+// API search type: parallel_for : (Io a, [a]) -> Io ()
+// Runs the function `f` in parallel on all the `idxs_to_iterate` container elements,
+// using `n_threads` threads.
+// The function will perform its side effects, and nothing is returned.
+template<typename F, typename ContainerIdxs>
+void parallel_for_n_threads(size_t n_threads, F f, const ContainerIdxs& idxs_to_iterate)
```

Comment:
What do you think about naming there functions
- `parallel_for_each`
- `parallel_for_each_n_threads`

instead?

It expresses the "execute for each element in this generic container", and also seems to be kind of a default name for such a function.

We could even introduce a normal (non-parallel) `fplus::for_each` too, so that it would align with the existing three `fplus::transform ...` functions. :slightly_smiling_face: